### ISSUE TITLE
Warning: Detected changes to resource object-store which is currently being deleted.

### ISSUE THREAD

**Issue by OPSTime** [rook/rook#7200]
Version：
kubernetes：v1.20.2
rook/ceph: v1.5.6  

$  kubectl apply -f object.yaml 
Warning: Detected changes to resource object-store which is currently being deleted.
cephobjectstore.ceph.rook.io/object-store configured

$ cat object.yaml |grep -v '#'
```
apiVersion: ceph.rook.io/v1
kind: CephObjectStore
metadata:
  name: object-store
spec:
  metadataPool:
    failureDomain: host
    replicated:
      size: 3
      requireSafeReplicaSize: true
    parameters:
      compression_mode: none
      target_size_ratio: ".5"
  dataPool:
    failureDomain: host
    replicated:
      size: 3
      requireSafeReplicaSize: true
    parameters:
      compression_mode: force
      target_size_ratio: ".5"
  preservePoolsOnDelete: false
  gateway:
    type: s3
    sslCertificateRef:
    port: 80
    instances: 1
    placement:
    annotations:
    labels:
    resources:
  healthCheck:
    bucket:
      disabled: false
      interval: 60s
    livenessProbe:
      disabled: false
```

**Comment by travisn** [comment]
@OPSTime If the object store CR is being deleted, K8s won't allow you to make changes to it. Did you intentionally delete the object store?

**Comment by OPSTime** [comment]
Yes, I want to recreate it。
What should I do?

**Comment by travisn** [comment]
@OPSTime You'll need to delete the object store CR before creating a new one. First check the operator log to see if there are any errors about deleting the object store. As a last resort, you can remove the finalizer on the CR as described [here](https://rook.github.io/docs/rook/v1.5/ceph-teardown.html#removing-the-cluster-crd-finalizer).

**Comment by OPSTime** [comment]
Thank you !
I found “finalizers” in CephObjectStore.

$ kubectl edit CephObjectStore -n rook-ceph
`apiVersion: ceph.rook.io/v1
kind: CephObjectStore
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"ceph.rook.io/v1","kind":"CephObjectStore","metadata":{"annotations":{},"name":"object-store","namespace":"rook-ceph"},"spec":{"dataPool":{"failureDomain":"host","parameters":{"compression_mode":"force","target_size_ratio":".5"},"replicated":{"requireSafeReplicaSize":true,"size":3}},"gateway":{"annotations":null,"instances":1,"labels":null,"placement":null,"port":80,"resources":null,"sslCertificateRef":null,"type":"s3"},"healthCheck":{"bucket":{"disabled":false,"interval":"60s"},"livenessProbe":{"disabled":false}},"metadataPool":{"failureDomain":"host","parameters":{"compression_mode":"none","target_size_ratio":".5"},"replicated":{"requireSafeReplicaSize":true,"size":3}},"preservePoolsOnDelete":false}}
  creationTimestamp: "2021-02-10T17:03:04Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2021-02-10T17:14:47Z"
  finalizers:
  - cephobjectstore.ceph.rook.io
  generation: 4
  name: object-store
  namespace: rook-ceph
  resourceVersion: "3908479"
  uid: 668070e8-9c05-4373-b477-197c092b5e32
spec:
`

**Comment by OPSTime** [comment]
i find error log from deploy/rook-ceph-operator:
2021-02-13 12:58:29.185118 E | ceph-object-controller: failed to reconcile failed to create object store deployments: failed to create object pools: failed to create data pool: failed to create pool object-store.rgw.buckets.data for object store object-store.: failed to create replicated pool object-store.rgw.buckets.data. Error ERANGE:  pg_num 32 size 3 would mean 819 total pgs, which exceeds max 750 (mon_max_pg_per_osd 250 * num_in_osds 3)
. : exit status 34

**Comment by OPSTime** [comment]
Error:
```
ERROR: Access to bucket 'ceph-bkt-e667551e-e0c9-482f-ab72-9cfa6e55b1eb' was denied
ERROR: S3 error: 403 (AccessDenied)
```

$ cat t.sh
```
object_svc=rook-ceph-rgw-object-store
bucket_name=ceph-delete-bucket
bucket_username=object-store-testuser
secret_name=`kubectl -n rook-ceph get CephObjectStoreUser/${bucket_username} -o jsonpath='{.status.info.secretName}'`
export AWS_HOST=$(kubectl -n default get cm ${bucket_name} -o jsonpath='{.data.BUCKET_HOST}')
export AWS_ACCESS_KEY_ID=$(kubectl -n rook-ceph get secret ${secret_name} -o jsonpath='{.data.AccessKey}' | base64 --decode)
export AWS_SECRET_ACCESS_KEY=$(kubectl -n rook-ceph get secret ${secret_name} -o jsonpath='{.data.SecretKey}' | base64 --decode)
export BUCKET=`kubectl get cm/ceph-delete-bucket -o jsonpath='{.data.BUCKET_NAME}'`


echo "export AWS_HOST=$AWS_HOST"
echo "export AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID"
echo "export AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY"
echo "export AWS_ENDPOINT=$object_svc"
echo "export BUCKET=$BUCKET"
```

$ bash t.sh
```
export AWS_HOST=rook-ceph-rgw-object-store.rook-ceph.svc
export AWS_ACCESS_KEY_ID=9O920A6XLBLRI0JNC8T5
export AWS_SECRET_ACCESS_KEY=sOhXqfLo7A0Sroq0x4DGlcx8GbyDaAHs4AAH232t
export AWS_ENDPOINT=rook-ceph-rgw-object-store
export BUCKET=ceph-bkt-e667551e-e0c9-482f-ab72-9cfa6e55b1eb
```

run export command
run "s3cmd ls --no-ssl --host=rook-ceph-rgw-object-store.rook-ceph.svc --host-bucket= s3://$BUCKET"
```
ERROR: Access to bucket 'ceph-bkt-e667551e-e0c9-482f-ab72-9cfa6e55b1eb' was denied
ERROR: S3 error: 403 (AccessDenied)
```

**Comment by travisn** [comment]
@OPSTime Can you summarize what issue you need help troubleshooting? Were you able to create the new object store or not? Now when creating a new pool, you're also reaching the PG limit. This could be increased through the toolbox.

**Comment by OPSTime** [comment]
I found the cause of the problem. The created user does not have the permission to access any of the buckets. It is necessary to set the policy to the bucket through the "s3cmd setpolicy" command to allow the user to access. Or modify the bucket owner through the "radosgw admin bucket link" command. The new user is not associated with any buckets.

**Comment by OPSTime** [comment]
@travisn  Do I have to manually associate users with buckets? When creating a bucket, can't I directly specify a user as the owner?

**Comment by travisn** [comment]
When a bucket is created with an OBC, a user is created for that bucket. If you want to add existing users to new buckets, then they wouldn't get access automatically.

**Comment by ningjunwei2** [comment]
@travisn  
![image](https://user-images.githubusercontent.com/88232846/182523192-205cab63-ddb6-47bf-b142-93bad797ed7f.png)
my cluser have the problem ,but i don't want to delete cluster , is anyway to fix  ?

**Comment by travisn** [comment]
See #10671
